devcontainer/postCreate.sh

```diff
@@ -7,7 +7,16 @@ set -euo pipefail
 #set R working directory
 ! grep -q "$1" $R_HOME/etc/Rprofile.site && sudo tee -a $R_HOME/etc/Rprofile.site <<< "setwd(\"$1\")"
 #set RStudio working directory
-! grep -q "$1" ~/.config/rstudio/rstudio-prefs.json && cat ~/.config/rstudio/rstudio-prefs.json | jq ". + {\"initial_working_directory\":\"$1\"}" >  ~/.config/rstudio/rstudio-prefs.json
+set_rstudio_pref() {
+    key="$1"
+    value="$2"
+    orig_prefs=$(jq -ne "input ? // {}" < ~/.config/rstudio/rstudio-prefs.json)
```

Comment:
We're using `jq` here, but we don't have jq installed in the research-template image. It's passing the tests because we're installing it specifically for them.

Could sed be used instead to insert  a line after the `{` ?

Reply:
jq isn't installed in this image, but is added by the process that starts up the dev container (amongst other stuff), which is why the test has to install it as it's working on the image alone but when the script runs in its intended environment it doesn't have to be explicitly installed. Bit confusing I know! `sed` could do the job of the two uses of `jq` in this script but I'd argue it'd be harder to read/understand .

Reply:
I didn't realise that jq was preinstalled. That is confusing! Do you have the list of packages that are preinstalled? Googling isn't helping me here.

The current use of `jq` is doing much more than a single `sed` command, I admit. In terms of trying to fix the bug, have you considered moving the R config to before the pip install? If it's a timing issue, then it might be worth trying to run it before the slow command.